compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -126,11 +126,6 @@ const HookSchema = z.object({
 
 export type Hook = z.infer<typeof HookSchema>;
 
-export const ModuleTypeResolver = z
-  .function()
-  .args(z.string())
-  .returns(z.nullable(TypeSchema));
-
```

Comment:
setting `.returns()` causes zod to return a function that internally validates the return type when you call it. that's cool but we want to control when the validation happens so i'm switching this to a plain function annotation